### ISSUE TITLE
TestBasicAuth.test_basic_invalid refers to Mock.assert_not_called that is not valid Mock assertion

### ISSUE THREAD

**Issue by swistakm** [talons/talons#25]
This Mock method doesn't exist so `assert_not_called` is another mock. This has no chance to raise any AssertionError ever:
https://github.com/talons/talons/blob/master/tests/auth/test_basicauth.py#L73

proof:

```
>>> import mock
>>> mock.Mock().assert_not_called()
<Mock name='mock.assert_not_called()' id='28293456'>
```

**Comment by jaypipes** [comment]
Nice catch :) Will fix up shortly!

**Comment by jaypipes** [comment]
Fixed in d5ac6403f6023d09f6e5553f8dfa28105b885180
